Assets/Scripts/World/ObjectBehaviors/LevelSwitchTrigger.cs

```diff
@@ -26,7 +29,11 @@ private void CheckIfValid(Collider other)
         if (other.gameObject.tag == activationTag)
         {
             if (other.gameObject.GetComponent<DynamicObject>().IsGrounded())
-                SceneManager.LoadScene(loadScene);
+            {
+                int nextScene = (loadNextScene) ? SceneManager.GetActiveScene().buildIndex + 1 : loadScene;
```

Comment:
This will fail if we have a number going out of bounds. I would set a modulo `SceneManager.sceneCountInBuildSettings` so that if we overflow we go back to level 0.

Reply:
I fixed it